src/psyclone/domain/lfric/kernel/meta_funcs_arg_metadata.py

```diff
@@ -0,0 +1,216 @@
+                 diff_basis_function=False):
+        self.function_space = function_space
+        # The setters for basis_function and diff_basis_function check
+        # for consistent values between the two. The
+        # diff_basis_function has not been set yet so the
+        # basis_function setter should not be used here.
+        self.check_boolean(basis_function, "basis_function argument")
+        self._basis_function = basis_function
+        self.diff_basis_function = diff_basis_function
+
+    def create_from_fparser2(fparser2_tree):
+        '''Create an instance of this class from an fparser2 tree.
+
+        :param fparser2_tree: fparser2 tree containing the metadata \
+            for a meta_funcs argument.
+        :type fparser2_tree: :py:class:`fparser.two.Fortran2003.Part_Ref`
+
+        :returns: an instance of this class.
+        :rtype: :py:class:`psyclone.domain.lfric.kernel.MetaFuncsArgMetadata`
+
+        '''
+        MetaFuncsArgMetadata.check_fparser2(
+            fparser2_tree, type_name="func_type")
+        nargs = MetaFuncsArgMetadata.get_nargs(fparser2_tree)
+        # There must be at least 2 and at most 3 arguments
+        if nargs < 2:
+            raise ParseError(
+                f"There must be at least 2 arguments, a function_space and "
+                f"one of basis_function or diff_basis_function, but found "
+                f"'{nargs}' arguments.")
+        if nargs > 3:
+            raise ParseError(
+                f"There must be at most 3 arguments, function_space, "
+                f"basis_function and diff_basis_function, but found "
+                f"'{nargs}'.")
+        function_space = MetaFuncsArgMetadata.get_arg(fparser2_tree, 0)
+
+        basis_function = False
+        diff_basis_function = False
+        arg1 = MetaFuncsArgMetadata.get_arg(fparser2_tree, 1)
+        if arg1.lower() == "gh_basis":
+            basis_function = True
+        elif arg1.lower() == "gh_diff_basis":
+            diff_basis_function = True
+        else:
+            raise ValueError(
+                f"The basis or differential basis value should be one of "
+                f"['gh_basis', 'gh_diff_basis'], but found '{arg1}'.")
+        arg2 = None
+        if nargs == 3:
+            arg2 = MetaFuncsArgMetadata.get_arg(fparser2_tree, 2)
+            if arg2.lower() == "gh_basis":
+                basis_function = True
+            elif arg2.lower() == "gh_diff_basis":
+                diff_basis_function = True
+            else:
+                raise ValueError(
+                    f"The basis or differential basis value should be one of "
+                    f"['gh_basis', 'gh_diff_basis'], but found '{arg2}'.")
+            if arg1.lower() == arg2.lower():
+                raise ParseError(
+                    f"The same basis or differential basis function value "
+                    f"should not be repeated, but found '{arg1}' twice.")
+        MetaFuncsArgMetadata._check_constraints(
```

Comment:
I've removed the check as it is not possible to get to that point with invalid values. However, you are right that it would be checked in the constructor anyway so was already redundant.